From caa45bd2cacce14d4cf6fb1a5fb0d3505a80c02c Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Thu, 24 Dec 2015 16:36:51 +0800
Subject: [PATCH 0796/5242] MLK-12075 ARM: dts: add imx7d-sdb-reva-touch.dtb

commit  756ccb25ba6d92c772fb3899af4d542aa80c1843 from
https://source.codeaurora.org/external/imx/linux-imx.git

For imx7d-sdb TO1.1 board, the touch interrupt pin no longer conflict
with HDMI, so move touch support is now inside imx7d-sdb.dts. To
compatible with TO1.0 board, still use imx7d-sdb-reva-touch.dts to
support touch.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                 |    1 +
 arch/arm/boot/dts/imx7d-sdb-reva-touch.dts |   14 +++++++++++++-
 arch/arm/boot/dts/imx7d-sdb-reva.dts       |    4 ++++
 arch/arm/boot/dts/imx7d-sdb.dts            |   10 ++++++++--
 4 files changed, 26 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 7db1227..dbba3da 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -617,6 +617,7 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sdb-reva-hdmi-audio.dtb \
 	imx7d-sdb-reva-m4.dtb \
 	imx7d-sdb-reva-qspi.dtb \
+	imx7d-sdb-reva-touch.dtb \
 	imx7d-sdb-reva-wm8960.dtb \
 	imx7s-colibri-eval-v3.dtb \
 	imx7s-warp.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
index 7c338d5..d3855e8d 100644
--- a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
@@ -6,4 +6,16 @@
  * published by the Free Software Foundation.
  */
 #include "imx7d-sdb-reva.dts"
-#include "imx7d-sdb-touch.dtsi"
+
+&sii902x {
+	status = "disabled";
+};
+
+&ecspi3 {
+	status = "okay";
+
+	tsc2046 {
+		interrupts = <13 0>;
+		pendown-gpio = <&gpio2 13 0>;
+	};
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb-reva.dts b/arch/arm/boot/dts/imx7d-sdb-reva.dts
index 6c47137..73e0b0b 100644
--- a/arch/arm/boot/dts/imx7d-sdb-reva.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-reva.dts
@@ -28,6 +28,10 @@
 	};
 };
 
+&ecspi3 {
+	status = "disabled";
+};
+
 &epdc {
 	pinctrl-0 = <&pinctrl_epdc0>;
 	en-gpios = <&extended_io 5 GPIO_ACTIVE_LOW>;
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 2f160ac..bb928aa 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -216,8 +216,9 @@
 };
 
 &ecspi3 {
+	fsl,spi-num-chipselects = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ecspi3>;
+	pinctrl-0 = <&pinctrl_ecspi3 &pinctrl_ecspi3_cs>;
 	cs-gpios = <&gpio5 9 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
@@ -772,7 +773,12 @@
 				MX7D_PAD_SAI2_TX_SYNC__ECSPI3_MISO	0x2
 				MX7D_PAD_SAI2_TX_BCLK__ECSPI3_MOSI	0x2
 				MX7D_PAD_SAI2_RX_DATA__ECSPI3_SCLK	0x2
-				MX7D_PAD_SD2_CD_B__GPIO5_IO9		0x59
+			>;
+		};
+
+		pinctrl_ecspi3_cs: ecspi3_cs_grp {
+			fsl,pins = <
+				MX7D_PAD_SD2_CD_B__GPIO5_IO9    0x80000000
 			>;
 		};
 
-- 
1.7.9.5

